### PR TITLE
State Management for bookmarks

### LINKED ISSUE #157: Implement bookmark of components and apis
Contributes to #136

### PR DESCRIPTION
**Pull request by chirag-singhal**
Contributes to #158 
Fixes #157 

## Changelog

Note - 

* Copied Async Storage
* State management for bookmarks
* Persisting bookmarks using async storage
* Add bookmark screen (Would need refactoring later, code copied from ExampleList)
* Add a button on Example List Screen to navigate to Bookmark screen (Need to remove this later)  

## Test Plan

Screenshot_2020-07-23-11-11-07-213_com rntester

Screenshot_2020-07-23-11-11-07-213_com rntester_1

Note - Haven't tested my changes on iOS (iOS build not working for most)

@AnshG714 can you try running it for iOS?

### CODE REVIEW

File: packages/RNTester/js/utils/AsyncStorage/RCTAsyncStorage.expo.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Expo managed apps don't include the @react-native-community/async-storage
```
Comment (stealthanthrax) [inline packages/RNTester/js/utils/AsyncStorage/RCTAsyncStorage.expo.js:2]: @chirag-singhal , do we need this file?
It is for the expo apps.

Comment (chirag-singhal) [reply packages/RNTester/js/utils/AsyncStorage/RCTAsyncStorage.expo.js:2]: @stealthanthrax yeah we don't require that. Removing that!

**Comment by manyaagarwal** [comment]
@YashKumarVerma can you check the metro.config file and see that it's linked to the React Native npm since I remember linking wasn't really working for you?

**Comment by YashKumarVerma** [comment]
> @YashKumarVerma can you check the metro.config file and see if that it is linked to the React Native npm package since I remember linking wasn't really working for you?

as of now, it's linked to the react-native npm package.
`const reactNativePath = path.resolve(__dirname, 'node_modules', 'react-native');`

**Comment by chirag-singhal** [comment]
@anku255 On iOS, it would give errors because we have removed the AsyncStorage. There are changes required in `RNTesterApp.ios.js` just like `RNTesterApp.android.js`.

**Comment by chirag-singhal** [comment]
@YashKumarVerma Strange, how's that side drawer with lists coming? @manyaagarwal  Have you tested these changes locally?

**Comment by chirag-singhal** [comment]
@YashKumarVerma we would remove the drawer #170, Can you try adding bookmarks from the main `ExampleList` Screen

**Comment by manyaagarwal** [comment]
@chirag-singhal Tested the changes. LGTM!

**Comment by chirag-singhal** [comment]
@anku255 I made changes for ios, but we can't test them now. So should we keep it on a hold or merge this?

**Pull request merged**